engine/runtime/src/test/scala/org/enso/compiler/test/pass/desugar/ImportsTest.scala

```diff
@@ -67,7 +67,7 @@ class ImportsTest extends CompilerTest {
       ir.imports.take(4).map(_.showCode()) shouldEqual List(
         "import Bar.Foo.Main as Foo",
         "import Bar.Foo.Main as Bar",
-        "from Bar.Foo.Main as Foo import Bar, Baz",
+        "from Bar.Foo.Main import Bar, Baz",
```

Comment:
Recently we discussed the imports/exports with @kazcw and I have to say I have no idea what `from Bar.Foo.Main as Foo import Bar, Baz` is supposed to mean?

Reply:
`from Bar.Foo.Main as Foo import Bar, Baz` should be pointless now as we shouldn't be bringing `Bar.Foo.Main` into scope.

Reply:
If you look into the test, the `.Main` is added to all `import X.Y` and `from X.Y` statements in a compiler phase.

Reply:
I accepted this case in the new parser because it's used in existing code, but if it makes no sense and we're eliminating it, I can remove support for the combination `from _ as _ import _`.

Reply:
I think that is correct. 

We should allow `import _ as _` though even that we are using a lot less now.

Reply:
Ok. Here are the files that need to be updated so I can remove support:
```
ubuntu@enso:~/repo/enso/wip$ grep 'from .* as .* import .*' $(find . -name '*.enso')
./test/Table_Tests/src/Delimited_Write_Spec.enso:from Standard.Table.Data.Column_Selector as Column_Selector_Module import By_Name
./test/Table_Tests/src/Delimited_Write_Spec.enso:from Standard.Table.Errors as Table_Errors import Column_Count_Mismatch_Data, Column_Name_Mismatch_Data
./test/Table_Tests/src/Excel_Spec.enso:from Standard.Table.Data.Column_Selector as Column_Selector_Module import By_Index
./test/Table_Tests/src/Excel_Spec.enso:from Standard.Table.Errors as Table_Errors import Invalid_Output_Column_Names_Data, Duplicate_Output_Column_Names_Data, Invalid_Location_Data, Range_Exceeded_Data, Existing_Data_Data, Column_Count_Mismatch_Data, Column_Name_Mismatch_Data
./test/Table_Tests/src/Table_Time_Spec.enso:from Standard.Table.Data.Data_Formatter as Data_Formatter_Module import Data_Formatter
./test/Table_Tests/src/Database/Common_Spec.enso:from Standard.Table.Data.Column_Selector as Column_Selector_Module import By_Name
./test/Table_Tests/src/Database/Common_Spec.enso:from Standard.Table.Errors as Table_Errors import No_Input_Columns_Selected, Missing_Input_Columns_Data
./test/Table_Tests/src/Table_Spec.enso:from Standard.Table.Errors as Table_Errors import Invalid_Output_Column_Names_Data, Duplicate_Output_Column_Names_Data, No_Input_Columns_Selected, Missing_Input_Columns_Data
./test/Table_Tests/src/Common_Table_Spec.enso:from Standard.Table.Errors as Table_Errors import all
./test/Table_Tests/src/Table_Time_Of_Day_Spec.enso:from Standard.Table.Data.Data_Formatter as Data_Formatter_Module import Data_Formatter
./test/Tests/src/System/File_Spec.enso:from Standard.Base.System.File.File_Permissions as File_Permissions_Module import all
./distribution/lib/Standard/Table/0.0.0-dev/src/Internal/Delimited_Writer.enso:from Standard.Table.Errors as Table_Errors import Duplicate_Output_Column_Names, Invalid_Output_Column_Names, Invalid_Row, Mismatched_Quote, Parser_Error, Additional_Invalid_Rows, Column_Count_Mismatch, Column_Name_Mismatch
./distribution/lib/Standard/Table/0.0.0-dev/src/Internal/Parse_Values_Helper.enso:from Standard.Table.Errors as Table_Errors import Invalid_Format_Data, Leading_Zeros_Data
```

Reply:
Removed them all in my PR #3701.

Reply:
So, removing support for those in the current parser is likely a waste of time (we are just more permissive than we should be). But I can make sure that later compiler passes (such as Imports) don't use rename when names are imported.

Reply:
It appears that we rely on this in a few places e.g., 
- https://github.com/enso-org/enso/blob/develop/distribution/lib/Standard/Base/0.0.0-dev/src/Meta.enso#L291
- https://github.com/enso-org/enso/blob/d8f274158a27fe821ace473dff6967acec18ac76/distribution/lib/Standard/Visualization/0.0.0-dev/src/Id.enso#L19
and a few other ones as well

Reply:
:crossed_fingers: my [last](https://github.com/enso-org/enso/pull/3710/commits/e936ef4f598e1959ed6ed75a57e8dc7932dadef4) commit covered all of those